bdb/rep.c

```diff
@@ -2843,25 +2848,25 @@ static void bdb_slow_replicant_check(bdb_state_type *bdb_state,
 static int bdb_track_replication_time(bdb_state_type *bdb_state,
                                       seqnum_type *seqnum, const char *host)
 {
-    struct waiting_for_lsn *waitforlsn;
+    if (!bdb_state->attr->track_replication_times)
+        return 0;
 
-    if (bdb_state->attr->track_replication_times) {
-        if (bdb_state->seqnum_info->waitlist[nodeix(host)] == NULL) {
-            bdb_state->seqnum_info->waitlist[nodeix(host)] =
-                malloc(sizeof(wait_for_lsn_list));
-            listc_init(bdb_state->seqnum_info->waitlist[nodeix(host)],
-                       offsetof(struct waiting_for_lsn, lnk));
-        }
+    struct waiting_for_lsn *waitforlsn;
+    int node_ix = nodeix(host);
+    if (bdb_state->seqnum_info->waitlist[node_ix] == NULL) {
+        bdb_state->seqnum_info->waitlist[node_ix] =
+            malloc(sizeof(wait_for_lsn_list));
+        listc_init(bdb_state->seqnum_info->waitlist[node_ix],
```

Comment:
Perhaps this malloc should be a calloc?

Reply:
Subsequent listc_init() sets all members of the listc_t so we don't need calloc here.